lib/publiccloud/provider.pm

```diff
@@ -379,12 +379,13 @@ sub vault_api {
     my $data   = $args{data}   // {};
     my $ua     = Mojo::UserAgent->new;
     my $url    = get_required_var('_SECRET_PUBLIC_CLOUD_REST_URL');
+    my $namespace = get_var('PUBLIC_CLOUD_VAULT_NAMESPACE', '');
     my $res;
 
     $self->vault_login() unless ($self->vault_token);
 
     $ua->insecure(get_var('_SECRET_PUBLIC_CLOUD_REST_SSL_INSECURE', 0));
-    $url = $url . $path;
+    $url = $url . '/v1/' . $namespace . $path;
```

Comment:
this PR is about adding namespace flexibility I get that , but how this correlate with 'v1' movements which is something different. 
Also actually I would change it to v2 actually . Such big change like adding namespace is worth it IMHO

Reply:
This v1 comes from `vault` rest API. 
Currently they only have `v1`. If we are going to get `v2` methods, I would say we add it as named parameter to `sub vault_api()` or just `vault_apiv2()`. But I don't see the need for now.

Reply:
Flexibility for the namespace mainly :-) But yes I could change this to `v2` as well as create a variable for this. @cfconrad, @jlausuch what do you think?

Reply:
I would leave it as it is, if you change it to `v2` it would break the communication with `vault` server, cause the URL just don't exists

Reply:
Ok, so let it as is for now and we will check this when a v2 will be available.